ui/app/services/permissions.js

```diff
@@ -0,0 +1,36 @@
+import Service, { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+// decide what happens when no policies are found
+// because requesting the root token's ACL policy always returns nil
+// https://github.com/hashicorp/vault/pull/4386/files#diff-a145099c8e3d917858619dcfb2ae09b9R3464
+
+export default Service.extend({
+  paths: null,
+  store: service(),
+
+  getPaths: task(function*() {
+    if (this.get('paths')) {
+      return;
+    }
+    let resp = yield this.get('store')
+      .adapterFor('permissions')
+      .query();
+    this.setPaths(resp);
+    return;
+  }),
+
+  setPaths(resp) {
+    this.set('paths', resp.data);
```

Comment:
Do you think we should store exact_paths and globbed_paths separately? This seems like a good place to check and see if it's a root token.